### DISCUSSION TITLE
Square Hole Magnet Bridges

### DISCUSSION THREAD

**Discussion by DavidCC00** [kennetek/gridfinity-rebuilt-openscad#62] [Ideas]
I'm not the original person to come up with this idea, but I am the one who made this printables page explaining the concept.  This is a case where a picture is worth a thousand words, so I'll just let the printables page do the explaining. 

https://www.printables.com/model/379547-gridfinity-magnet-holes-model-for-example-and-tink

**Comment by Ruudjhuu** [comment]
The idea is cool. It is definitely something I am going to use when I need to print big holes.

I do think that the holes for magnets in gridfinity are small enough to be able to print with one "square". This repo already contains a method to overcome this issue. It is explained in the README under `printable holes` . You can toggle this whit the `style_hole` variable. Pick `magnet and screw holes - printable slit`. The method is explained in [this video](https://www.youtube.com/watch?v=W8FbHTcB05w)

**Reply by DavidCC00** [reply]
Printable slit accomplishes a similar thing, but I don't want screw holes.  They add print time/filament and reduces the depth a bin design can be.

**Reply by Ruudjhuu** [reply]
The "magnet holes only" feature does not require a circle printed in the air. The layer on top of a magnet hole reach from side to side in a straight line. You should not need a slit to print a bin with magnet holes. The printer should be able to overcome this gap without much effort by just bridging. If you are not able to bridge this small gap with your printer, I would advise you to create a post on [r/FixMyPrint](https://www.reddit.com/r/FixMyPrint/). This community can give you good tips to be able to make the quality of your bridges better.

I also did take a second look at your link and checked some of your models with the magnets+ feature. I honestly do not see any improvement in print-ability for this kind of holes. 

* The first line you print from the biggest square already almost bridges the whole hole. Why do you think it is not possible to make that extra 1.91 mm? and what is the added value of the smaller squares if your initial bridge is already twice the size? The only "added" value I see here is that your glued magnet won't stick as well. With larger holes, I can see it working as the difference is bigger. It could also be improved if you use a pentagon, or even a hexagon. 
![slit](https://user-images.githubusercontent.com/31508944/217905989-12285862-ee1f-4add-afb9-94d3a50b0169.png)

* The corner of the biggest square does not even touch the edge of the circle. Depending on your nozzle size this would make things even worse than just bridging the hole. However, this is easy to fix.
![slit](https://user-images.githubusercontent.com/31508944/217906860-4361a0d8-0275-41bb-a962-5a5efb77d27c.png)

**Comment by kennetek** [comment]
I agree with @Ruudjhuu. I think its cool and would work better for larger holes, but I think the single slit is enough for this application.
